File: Studia/Projekt_Hurtowni/src/projekt_hurtowni/Magazyn.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package projekt_hurtowni;

import java.util.*;
/**
 *
 * @author Bartek
 */
public class Magazyn {
    String adres;
    List<Produkt> Produkty = new ArrayList<Produkt>();
    List<Integer> ilosci = new ArrayList<Integer>();
    /**
     *
     * @param n
     */
    public Magazyn(String n)
    {
        adres = n;
    }
    public void dodaj_produkt(Produkt p, int i)
    {
        if(czy_na_liscie(p))
            System.out.print("Obiekt juz w magazynie");
        else
        {
            Produkty.add(p);
            ilosci.add(i);
        }
    }
    public void usun_produkt(Produkt p)
    {
        if(czy_na_liscie(p))
        {
            int indeks = Produkty.indexOf(p);
            Produkty.remove(p);
            ilosci.remove(indeks);
        }
        else
        {
            System.out.print("Obiektu nie ma w magazynie");
        }
    }
    public void wyswietl_magazyn()
    {
        System.out.println("Adres Magazynu: " + adres);
        System.out.println("Produkty na liscie: ");
        int indeks = 0;
        while(indeks < Produkty.size())
        {
            Produkty.get(indeks).wyswietl_produkt();
            System.out.print(", ilosc: " + ilosci.get(indeks));
            System.out.println();
            indeks++;
        }
        
    }
    public Boolean czy_na_liscie(Produkt p)
    {
        return Produkty.indexOf(p) != -1;
    }
    public void edytuj_ilosc(Produkt p, int i)
    {
        if(czy_na_liscie(p))
        {
            int indeks = Produkty.indexOf(p);
            ilosci.set(indeks,i);
        }
        else System.out.print("Obiektu nie ma w magazynie");
    }
    
}

File: Studia/Projekt_Hurtowni_c++/Magazyn.cpp
#include "Classes/Magazyn.hpp"

Magazyn::Magazyn(string n) {top = nullptr; adres=n;}

Magazyn::~Magazyn()
{
	while(top != nullptr)
	{
		list* temp = new list;
		temp = top;
		top = top->next;
		delete temp;
	}
}
		
void Magazyn::dodaj_produkt(Produkt* p, int i, int n)
{
	if(!sprawdz_czy_na_liscie(p))
	{
		list* l = new list;
		l->prod = p;
		l->ilosc = i;
		l->nr_prod = n;
		
		if(top == nullptr) top = l;
		else
		{
			l->next = top;
			top->prev = l;
			top = l;	
		}
	}
	else
	{
		cout<<"Error::Produkt juz jest na liscie."<<endl;
	}
	
}

void Magazyn::usun_produkt(int n)
{
	if (top==nullptr) cout<<"Nie ma juz zadnych produktow na liscie."<<endl;
	else
	{
		list* l = new list;
		l = top;
		bool flaga=1;
		while(l!=nullptr && flaga)
		{
			if(l->nr_prod==n)
			{
				if(l->prev==nullptr && l->next==nullptr) top = nullptr;
				else if(l->prev!=nullptr && l->next==nullptr) l->prev->next = nullptr;
				else if(l->prev==nullptr && l->next!=nullptr) 
				{
					top = l->next;
					l->next->prev = nullptr;
				}
				else if(l->prev!=nullptr && l->next!=nullptr)
				{
					l->prev->next = l->next;
					l->next->prev = l->prev;
				}
				flaga = 0;
			}
			if(flaga) l = l->next;
		}
		if(flaga==1) cout<<"Produktu nie bylo na liscie."<<endl;
		else delete l;
	}
}

void Magazyn::wyswietl_magazyn()
{
	cout<<"Adres Magazynu: "<<adres<<endl<<"Lista Przechowywanych produktow:"<<endl;
	list* l = new list;
	l = top;
	while(l!=nullptr)
	{
		cout<<"nr."<<l->nr_prod<<" ";
		l->prod->wyswietl_produkt();
		cout<<", ilosc w magazynie: "<<l->ilosc;
		cout<<endl;
		l = l->next;
	}
}

bool Magazyn::sprawdz_czy_na_liscie(Produkt* p)
{
	list* l = new list;
	l = top;
	while(l!=nullptr)
	{
		if(p == l->prod) return 1;
		l = l->next;
	}
	return 0;
}

void Magazyn::edytuj_ilosc(Produkt* p, int l)
{
	bool flaga=1;
	list* i = new list;
	i = top;
	while(i!=nullptr)
	{
		if(p == i->prod) 
		{
			i->ilosc = l;
			flaga = 0;
		}
		i = i->next;
	}
}
